test/unit/org/apache/cassandra/tools/nodetool/stats/NodetoolTableStatsTest.java

```diff
@@ -0,0 +1,237 @@
+                        "            local_read_latency_ms, local_write_latency_ms,\n" + 
+                        "            maximum_live_cells_per_slice_last_five_minutes,\n" + 
+                        "            maximum_tombstones_per_slice_last_five_minutes, memtable_cell_count,\n" + 
+                        "            memtable_data_size, memtable_off_heap_memory_used,\n" + 
+                        "            memtable_switch_count, number_of_partitions_estimate,\n" + 
+                        "            off_heap_memory_used_total, pending_flushes, percent_repaired,\n" + 
+                        "            read_latency, reads, space_used_by_snapshots_total, space_used_live,\n" + 
+                        "            space_used_total, sstable_compression_ratio, sstable_count,\n" + 
+                        "            table_name, write_latency, writes)\n" + 
+                        "\n" + 
+                        "        -t <top>, --top <top>\n" + 
+                        "            Show only the top K tables for the sort key (specify the number K of\n" + 
+                        "            tables to be shown\n" + 
+                        "\n" + 
+                        "        -u <username>, --username <username>\n" + 
+                        "            Remote jmx agent username\n" + 
+                        "\n" + 
+                        "        --\n" + 
+                        "            This option can be used to separate command-line options from the\n" + 
+                        "            list of argument, (useful when arguments might be mistaken for\n" + 
+                        "            command-line options\n" + 
+                        "\n" + 
+                        "        [<keyspace.table>...]\n" + 
+                        "            List of tables (or keyspace) names\n" + 
+                        "\n" + 
+                        "\n";
+        Assertions.assertThat(tool.getStdout()).isEqualTo(help);
+    }
+
+    @Test
+    public void testTableStats()
+    {
+        ToolResult tool = ToolRunner.invokeNodetool("tablestats");
+
+        assertThat(tool.getStdout(), CoreMatchers.containsStringIgnoringCase("Keyspace : system_schema"));
+        assertTrue(StringUtils.countMatches(tool.getStdout(), "Table:") > 1);
+        assertTrue(tool.getCleanedStderr().isEmpty());
+        assertEquals(0, tool.getExitCode());
+
+        tool = ToolRunner.invokeNodetool("tablestats", "system_distributed");
+        assertThat(tool.getStdout(), CoreMatchers.containsStringIgnoringCase("Keyspace : system_distributed"));
+        assertThat(tool.getStdout(), CoreMatchers.not(CoreMatchers.containsStringIgnoringCase("Keyspace : system_schema")));
+        assertTrue(StringUtils.countMatches(tool.getStdout(), "Table:") > 1);
+        assertTrue(tool.getCleanedStderr().isEmpty());
+        assertEquals(0, tool.getExitCode());
+    }
```

Comment:
It seems that the pattern run tool - check stdout - check stderr - check exit code is repeated for most tests. Not saying that we should do it given that the tests are simpler enough, but we could use a common utility method like:
```suggestion
    private static void assertToolSuccess(Consumer<String> stdoutCheck, String... args)
    {
        ToolResult tool = ToolRunner.invokeNodetool(args);
        stdoutCheck.accept(tool.getStdout());
        assertTrue(tool.getCleanedStderr().isEmpty());
        assertEquals(0, tool.getExitCode());
    }

    @Test
    public void testTableStats()
    {
        assertToolSuccess(stdout -> {
            assertTrue(stdout.contains("Keyspace : system_schema"));
            assertTrue(StringUtils.countMatches(stdout, "Table:") > 1);
        }, "tablestats");

        assertToolSuccess(stdout -> {
            assertTrue(stdout.contains("Keyspace : system_distributed"));
            assertFalse(stdout.contains("Keyspace : system_schema"));
            assertTrue(StringUtils.countMatches(stdout, "Table:") > 1);
        }, "tablestats", "system_distributed");
    }
```
That's just a suggestion, feel free to ignore.

Reply:
Multiple checks on stdout are not the norm but the exception on the other many tooling tests we have just been introducing. So I see little added value and I would have to change a gazillion other tests to make things consistent. I would rather leave it as it is if you don't mind.

Reply:
Well, that would remove a gazillion other code duplications :). I understand that's why `ToolRunner#assertOnCleanExit` was added during CASSANDRA-15942 (in slightly different way to its current state), I propose we use that method instead of the two assertions. I can count 12 places where it can be used, so not that many.